src/v/lsm/core/internal/files.h

```diff
@@ -36,10 +37,54 @@ consteval file_id operator""_file_id(unsigned long long val) {
     return file_id{static_cast<uint64_t>(val)};
 }
 
+// Database epoch are assigned to a file such that on shared storage mediums
+// databases that share persistence locations that write with different epochs
+// will not collide with each other.
```

Comment:
how do you define "collision" when it pertains to garbage collection? IIUC collision here is primarily about new files that are created: a leader with epoch/term N may be operating in parallel with an isolated leader operating at epoch/term N-1. all new files will be created with the respective, non-colliding terms and only the manifest.N is the one that really counts. but what about for GC? how do we prevent collisions from deletes by isolated leaders?

Reply:
Yeah for GC I settled we have to have some delay to reduce the likely hood of deletes by isolated leaders being an issue. We should talk about this more, @andrwng and I have chatted a bit